src/schema/browser-apis.js

```diff
@@ -11,46 +11,68 @@ const schemas = schemaList.reduce(
   {}
 );
 
-export function isDeprecatedApi(namespace, property) {
-  return DEPRECATED_APIS.includes(`${namespace}.${property}`);
+function getObjectProperty(schema, property) {
+  for (const schemaProperty of schemaObjectNames) {
+    if (
+      schema &&
+      schema[schemaProperty] &&
+      property in schema[schemaProperty]
+    ) {
+      return schema[schemaProperty][property];
+    }
+  }
+  return null;
+}
+
+function getArrayProperty(schema, property) {
+  for (const schemaProperty of schemaArrayNames) {
+    if (schema && schemaProperty in schema) {
+      const namespaceProperties = schema[schemaProperty];
+      if (Array.isArray(namespaceProperties)) {
+        for (const schemaItem of namespaceProperties) {
+          if (schemaItem.name === property) {
+            return schemaItem;
+          }
+        }
+      }
+    }
+  }
+  return null;
 }
 
 export function isTemporaryApi(namespace, property) {
   return TEMPORARY_APIS.includes(`${namespace}.${property}`);
 }
 
-function hasObjectProperty(schema, property) {
-  return schemaObjectNames.some((schemaProperty) => {
-    return schema[schemaProperty] && property in schema[schemaProperty];
-  });
-}
+export function isDeprecatedApi(namespace, property) {
+  const schema = schemas[namespace];
 
-function hasArrayProperty(schema, property) {
-  return schemaArrayNames.some((schemaProperty) => {
-    const namespaceProperties = schema[schemaProperty];
-    return (
-      Array.isArray(namespaceProperties) &&
-      namespaceProperties.some((schemaItem) => {
-        return schemaItem.name === property;
-      })
-    );
-  });
+  const schemaItem =
+    getObjectProperty(schema, property) || getArrayProperty(schema, property);
+
+  return (
+    schemaItem !== null &&
+    schemaItem.deprecated !== undefined &&
+    DEPRECATED_JAVASCRIPT_APIS.includes(`${namespace}.${property}`)
+  );
 }
 
 export function hasBrowserApi(namespace, property) {
   const schema = schemas[namespace];
-  // We "have" the API if it's deprecated or temporary so
-  // we don't double warn.
+  // We "have" the API if it's deprecated or temporary so we don't double warn.
   if (
-    isDeprecatedApi(namespace, property) ||
-    isTemporaryApi(namespace, property)
+    isTemporaryApi(namespace, property) ||
+    isDeprecatedApi(namespace, property)
   ) {
     return true;
   }
-  if (!schema) {
+
+  const schemaItem =
+    getObjectProperty(schema, property) || getArrayProperty(schema, property);
+
+  if (schemaItem !== null && schemaItem.unsupported !== undefined) {
     return false;
   }
-  return (
-    hasObjectProperty(schema, property) || hasArrayProperty(schema, property)
-  );
+
+  return schemaItem !== null;
```

Comment:
It is true that [when a schema item contains a defined `unsupported` property then it is also set to true](https://searchfox.org/mozilla-central/search?q=%22unsupported%22%3A&case=false&regexp=false&path=schemas%2F), but given that it also always a boolean we may use `... && schemaItem.unsupported) {...}` here.

It may also good to extend the check to also verify that `schemaItem` is not `undefined` (e.g. by just changing `schemaItem !== null` into `schemaItem != null`).

Could we turn these checks (from line 73 to line 77) into a single line return statement?

```
return schemaItem && !schemaItem.unsupported;
```